File: LC/lc53/lc53.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxCont, maxSoFar = -1000000, -1000000
        for c in nums:
            maxCont = max(c, maxCont + c)
            maxSoFar = max(maxCont, maxSoFar)
        return maxSoFar

File: LC/lc53/lc53.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
  var cur = nums[0], res = nums[0];
  for (i = 1; i < nums.length; i++){
      cur = Math.max(cur + nums[i], nums[i]);
      res = Math.max(cur, res);
  }
  return res;
};